packages/inference/src/snippets/getInferenceSnippets.ts

```diff
@@ -136,54 +175,71 @@ const snippetGenerator = (templateName: string, inputPreparationFn?: InputPrepar
 			},
 			model,
 			provider,
 			providerModelId: providerModelId ?? model.id,
 		};
 
-		/// Iterate over tools => check if a snippet exists => generate
-		return TOOLS.map((tool) => {
-			if (!hasTemplate(tool, templateName)) {
-				return;
-			}
-			const template = loadTemplate(tool, templateName);
-			if (tool === "huggingface_hub" && templateName === "basic") {
-				if (!(model.pipeline_tag && model.pipeline_tag in HFH_INFERENCE_CLIENT_METHODS)) {
-					return;
-				}
-				params["methodName"] = HFH_INFERENCE_CLIENT_METHODS[model.pipeline_tag];
-			}
+		/// Iterate over clients => check if a snippet exists => generate
+		return inferenceSnippetLanguages
+			.map((language) => {
+				return CLIENTS[language]
+					.map((client) => {
+						if (!hasTemplate(language, client, templateName)) {
+							return;
+						}
+						const template = loadTemplate(language, client, templateName);
+						if (client === "huggingface_hub" && templateName.includes("basic")) {
+							if (!(model.pipeline_tag && model.pipeline_tag in HF_PYTHON_METHODS)) {
+								return;
+							}
+							params["methodName"] = HF_PYTHON_METHODS[model.pipeline_tag];
+						}
 
-			/// Generate snippet
-			let snippet = template(params).trim();
-			if (!snippet) {
-				return;
-			}
+						if (client === "huggingface.js" && templateName.includes("basic")) {
+							if (!(model.pipeline_tag && model.pipeline_tag in HF_JS_METHODS)) {
+								return;
+							}
+							params["methodName"] = HF_JS_METHODS[model.pipeline_tag];
+						}
 
-			/// Add import section separately
-			if (tool === "huggingface_hub") {
-				const importSection = snippetImportInferenceClient({ ...params });
-				snippet = `${importSection}\n\n${snippet}`;
-			} else if (tool === "requests") {
-				const importSection = snippetImportRequests({
-					...params,
-					importBase64: snippet.includes("base64"),
-					importJson: snippet.includes("json."),
-				});
-				snippet = `${importSection}\n\n${snippet}`;
-			}
+						/// Generate snippet
+						let snippet = template(params).trim();
+						if (!snippet) {
+							return;
+						}
+
+						/// Add import section separately
+						if (client === "huggingface_hub") {
+							const importSection = snippetImportPythonInferenceClient({ ...params });
+							snippet = `${importSection}\n\n${snippet}`;
+						} else if (client === "requests") {
+							const importSection = snippetImportRequests({
+								...params,
+								importBase64: snippet.includes("base64"),
+								importJson: snippet.includes("json."),
+							});
+							snippet = `${importSection}\n\n${snippet}`;
+						}
 
-			/// Snippet is ready!
-			return { client: tool, content: snippet };
-		}).filter((snippet) => snippet !== undefined && snippet.content) as InferenceSnippet[];
+						/// Snippet is ready!
+						return { language: language, client: client, content: snippet };
+					})
+					.filter((snippet) => snippet !== undefined && snippet.content) as InferenceSnippet[];
```

Comment:
Let's use a type predicate
```suggestion
					.filter((snippet): snippet is InferenceSnippet => snippet);
```

Reply:
:exploding_head: